spec/json-api-vanilla/diff_spec.rb

```diff
@@ -94,6 +94,35 @@
     end.to raise_error(JSON::Api::Vanilla::InvalidRootStructure)
   end
 
+  it "should raise an error if the document contains an unrecognized element" do
+    json = <<-JSON
+    {
+      "type": "object",
+      "id": "123",
+      "attributes": {},
+      "foo": "bar"
```

Comment:
This test only fails because it doesn't have a toplevel element with a `data`, `errors` or `meta` field, right?

Rather than because it contains an unrecognized element.

Reply:
Correct. That's why I called the method naive, like the top-level validation is named. I think rejecting unknown fields across the entire structure would be a good idea (and is actually the correct behavior according to the json:api spec) but that would be inconsistent with the existing validation, and that would be a pervasive enough change to warrant more discussion.

Reply:
OK. Could you make the test description a bit more accurate? Something like "should raise an error if the document lacks a required field".